edge-modules/DirectMethodCloudSender/src/Program.cs

```diff
@@ -90,17 +83,15 @@ static async Task<int> MainAsync()
 
                         if (result.Status == (int)HttpStatusCode.OK)
                         {
-                            var eventMessage = new Message(Encoding.UTF8.GetBytes($"Direct Method [{transportType}] Call succeeded."));
-                            eventMessage.Properties.Add("sequenceNumber", count.ToString());
-                            eventMessage.Properties.Add("batchId", batchId.ToString());
-                            Logger.LogInformation($"Calling Direct Method from cloud with count {count} succeeded.");
-                            await moduleClient.SendEventAsync(RouteOutputName, eventMessage);
+                            Logger.LogDebug($"Calling Direct Method from cloud with count {count} returned with status code {result.Status}.");
                         }
                         else
                         {
                             Logger.LogError($"Calling Direct Method from cloud with count {count} failed with status code {result.Status}.");
                         }
 
+                        CallAnalyzerToReportStatus(moduleId, result, analyzerClient);
```

Comment:
we should report if there is any exception and treat as a failed call.

Reply:
The CallAnalyzerToReportStatus function will log the exception. What do you mean by treat as a failed call?

Reply:
As discussed, this should increment exception action and failed counts.

Reply:
Turns out I was already doing this

Reply:
you need to report exception to analyzer in below catch blocks.

Reply:
We can add the exception reporting as part of offline, is not required to be added with this commit